### PR TITLE
Pricing in server

### PR DESCRIPTION
**Pull request by davidfischer**
**Note:** This is branched from the Django 4 upgrade: https://github.com/readthedocs/ethical-ad-server/pull/781.

Allow advertisers to select regions and topics when requesting a flight. Based on what they specify, an estimated CPM will be shown. Prices per region/topic will be stored in a JSON field on the region model. Regions and topics will have an option whether they can be selected by advertisers when requesting a new flight.

* Currently when selecting multiple regions and topics, an average is taken. So if there's 2 regions selected and 2 topics selected, the price will be an average of the 4 combinations. This seems a bit awkward.
* Discounts are currently hard coded

## Screenshot

Screenshot from 2023-08-23 16-55-45

### CODE REVIEW

File: adserver/validators.py [hunk 1]
```diff
@@ -7,6 +7,46 @@
...
+    code = "topic-pricing-validator"
+    limit_value = None  # required for classes extending BaseValidator
+
+    def __init__(self, message=None):
+        """Initialization for the topic pricing validator."""
+        if message:
+            self.message = message
+
+    def __call__(self, value):
+        from .models import Topic  # noqa
```
Comment (ericholscher) [inline adserver/validators.py:25]: Guessing there's a reason this is a nested import, but probably good to note it here?

File: assets/src/views/flight-request.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const ko = require('knockout');
+import { getFlightPrice, getDiscount } from "../libs/flight-utils";
+
+
+function FlightRequestViewModel(method) {
+  this.budget = ko.observable($("#id_budget").val());
+  this.regions = ko.observable([...document.querySelectorAll("form [name='regions']:checked")].map(function (node) { return node.value }));
+  this.topics = ko.observable([...document.querySelectorAll("form [name='topics']:checked")].map(function (node) { return node.value }));
+
+  this.pricing = JSON.parse($('#data-pricing').text());
```
Comment (ericholscher) [inline assets/src/views/flight-request.js:10]: Is this used? Seems like we're loading pricing from there in `getFlightPrice` also?

**Review by ericholscher — APPROVED** [review]
Excited for this 🎉

File: assets/src/views/flight-request.js (hunk 2 above)
Comment (davidfischer) [reply assets/src/views/flight-request.js:10]: No. It can be removed.

File: adserver/validators.py (hunk 1 above)
Comment (davidfischer) [reply adserver/validators.py:25]: The validators are used to validate the models. It would be circular without this. I'll note it.

**Pull request merged**